### ISSUE TITLE
No Notification

### ISSUE THREAD

**Issue by jagsus** [silviokennecke/homebridge-gpio-doorbell#2]
Hi,

I have followed your instructions to trigger a webhook. When pressing the Button, Homebridge gives the following output:
```
[13/09/2022, 13:31:11] [KlingelKnopf] Doorbell "KlingelKnopf" rang.
[13/09/2022, 13:31:11] [KlingelKnopf] Performing request to webhook at http://127.0.0.1:8087/doorbell?KlingelKamera.
```
The camera itself worked fine (until adding the doorbell option). Now with your plugin added, when pressing the button nothing happens, except the above output by homebridge. I have attached my config, hoping you can help me out!

```
"accessories": [
        {
            "name": "KlingelKnopf",
            "gpioPin": 11,
            "throttleTime": 1000,
            "negateInput": false,
            "enableOutput": false,
            "enableHttpTrigger": true,
            "httpTriggerUrl": "http://127.0.0.1:8087/doorbell?KlingelKamera",
            "accessory": "GpioDoorbell"
        }
    ],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "name": "Camera FFmpeg",
            "porthttp": 8087,
            "cameras": [
                {
                    "name": "KlingelKamera",
                    "doorbell": true,
                    "unbridge": true,
                    "videoConfig": {
                        "source": "-i rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mp4"
                    }
                }
            ],
            "platform": "Camera-ffmpeg"
        }
    ]
```

**Comment by silviokennecke** [comment]
Hi @jagsus,
could you please send me the output of the following commands?

```
netstat -tulpen
curl -v http://127.0.0.1:8087/doorbell?KlingelKamera
```

**Comment by jagsus** [comment]
Sure! 

`netstat -tulpen` outputs:
```
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode                                                                                                                 PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      0          14424                                                                                                                 -
tcp        0      0 127.0.0.1:6010          0.0.0.0:*               LISTEN      1000       14310                                                                                                                 -
tcp6       0      0 :::35503                :::*                    LISTEN      110        13677                                                                                                                 -
tcp6       0      0 :::8888                 :::*                    LISTEN      0          12607                                                                                                                 -
tcp6       0      0 :::22                   :::*                    LISTEN      0          14435                                                                                                                 -
tcp6       0      0 ::1:6010                :::*                    LISTEN      1000       14309                                                                                                                 -
tcp6       0      0 :::8581                 :::*                    LISTEN      110        13665                                                                                                                 -
tcp6       0      0 :::8087                 :::*                    LISTEN      110        12847                                                                                                                 -
tcp6       0      0 :::51694                :::*                    LISTEN      110        13679                                                                                                                 -
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           110        13662                                                                                                                 -
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           108        13486                                                                                                                 -
udp        0      0 0.0.0.0:50939           0.0.0.0:*                           108        13488                                                                                                                 -
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          12008                                                                                                                 -
udp6       0      0 :::5353                 :::*                                108        13487                                                                                                                 -
udp6       0      0 :::47852                :::*                                108        13489                                                                                                                 -

```

`curl -v http://127.0.0.1:8087/doorbell?KlingelKamera` outputs:
```
*   Trying 127.0.0.1:8087...
* Connected to 127.0.0.1 (127.0.0.1) port 8087 (#0)
> GET /doorbell?KlingelKamera HTTP/1.1
> Host: 127.0.0.1:8087
> User-Agent: curl/7.74.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 14 Sep 2022 08:24:53 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
< Transfer-Encoding: chunked
<
* Connection #0 to host 127.0.0.1 left intact
{"error":false,"message":"Doorbell switched on."}
```

Hope this helps :)

**Comment by silviokennecke** [comment]
When calling the curl, did you get a notification in HomeKit (push notification, doorbell on HomePod, …)?

**Comment by silviokennecke** [comment]
When triggering a webhook, the doorbell provided by this plugin will not work anymore. This is because it is assumed that in this case the hombridge-ffmpeg plugin does that. If the doorbell doesn’t ring when running the curl command, I sent you, the issue will probably be the ffmpeg plug-in or your HomeKit configuration

**Comment by jagsus** [comment]
No, nothing happened when calling the curl. I have not done any configuring in the Home App. Just added the Homebridge. What I noticed was that after adding your plugin and the webhook, the camera disappeared from the Home App.

**Comment by jagsus** [comment]
Out of curiosity I just removed the ffmpeg part from my config and the httpTrigger lines from your plugins config. Homebridge now says:
```
[14/09/2022, 11:43:06] [KlingelKnopf] Doorbell "KlingelKnopf" rang.
[14/09/2022, 11:43:06] [KlingelKnopf] Forwarding ring directly to HomeKit.
```

Still, nothing happens. I have tried removing the Bridge from the Home App and resetting homebridge with no effect. Is this normal behaviour because HomeKit expects a camera?

```
"accessories": [
        {
            "name": "KlingelKnopf",
            "gpioPin": 11,
            "throttleTime": 1000,
            "negateInput": false,
            "enableOutput": false,
            "accessory": "GpioDoorbell"
        }
    ],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        }
    ]
```

**Comment by silviokennecke** [comment]
The accessory exposed by the doorbell plugin is in some cases not directly visible in the HomeKit app from Apple, as the official specification assumes that the doorbell characteristic is exposed by a camera accessory. But it works anyway.
If the ring isn't forwarded, even with only the doorbell plugin active, I'd check if the doorbell is allowed to send notifications. 

I would add the ffmpeg plugin again and check the notification settings in the camera accessory.

![IMG_67B08FF8E3C2-1](https://user-images.githubusercontent.com/17751219/190179073-db3a7d51-822a-46a5-868e-eb02385bc5bd.jpeg)

**Comment by jagsus** [comment]
Okay, I misunderstood that. I have now reinstalled basically everthing, updated to iOS16 and still get the same result (no notification). Here are some screenshots from the home app, as well as my homebridge config. If you don't see anything weird here, I'll try my luck with the camera-ffmepg plugin devs.

https://imgur.com/a/MVVqdQK

```
 "accessories": [
        {
            "name": "KlingelKnopf",
            "gpioPin": 11,
            "throttleTime": 1000,
            "negateInput": false,
            "enableOutput": false,
            "enableHttpTrigger": true,
            "httpTriggerUrl": "http://127.0.0.1/doorbell?KlingelKamera",
            "accessory": "GpioDoorbell"
        }
    ],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "name": "Camera FFmpeg",
            "porthttp": 8080,
            "cameras": [
                {
                    "name": "KlingelKamera",
                    "doorbell": true,
                    "unbridge": true,
                    "videoConfig": {
                        "source": "-i rtsp://rtsp.stream/pattern"
                    }
                }
            ],
            "platform": "Camera-ffmpeg"
        }
    ]
```

**Comment by silviokennecke** [comment]
From what you wrote it more looks like a general issue either with homebridge, HomeKit or your network.

What you could try is enabling the debug mode in homebridge and try again. Probably some debug messages explain, what is going on here. To run homebridge in debug mode simply add `-D` as flag.

**Comment by jagsus** [comment]
Okay, it's working now... It was a dumb mistake in my network topology...

The router (fritzbox) provided by our Internet Provider has very poor Wifi, so I disabled its Wifi and hooked up another Wifi Router (zyxel) to the Fritzbox to manage Wifi. This has worked great so far. Now I had my Apple TV hooked up to the Fritzbox and everything else via Wifi to the Zyxel. That seemed to work... the Home App recognized the Apple TV as Home Hub, I got no errors so I thought this would be fine...
I just hooked up the Apple TV to the Zyxel and... voila... everything works as expected. 

Thank you very much for your help :)
